src/transformers/core_model_loading.py

```diff
@@ -122,127 +118,134 @@ def convert(
     ) -> dict[str, list[torch.Tensor]]:
         raise NotImplementedError
 
+    @property
+    def reverse_op(self) -> ConversionOps:
+        raise NotImplementedError
+
 
 class Chunk(ConversionOps):
-    """Split a tensor along ``dim`` into equally sized chunks or using explicit ``sizes``."""
-
-    reverse_op: type[ConversionOps]
+    """Split a tensor along ``dim`` into equally sized chunks."""
 
-    def __init__(self, dim: int = 0, chunks: Optional[int] = None, sizes: Optional[Sequence[int]] = None):
+    def __init__(self, dim: int = 0):
         self.dim = dim
-        self.chunks = chunks
-        self.sizes = list(sizes) if sizes is not None else None
-        self.reverse_op = Concatenate
 
     def convert(
         self,
         value: dict[str, list[torch.Tensor]],
-        source_keys: list[str],
-        target_keys: list[str],
+        source_patterns: list[str],
+        target_patterns: list[str],
         full_layer_name: str,
         model,
         missing_keys,
         config,
     ) -> dict[str, list[torch.Tensor]]:
-        tensors = next(iter(value.values()))
-        tensor = tensors[0]
-        sizes = len(target_keys)
+        if len(value) > 1:
+            raise ValueError("Unexpected value in `Chunk`!")
```

Comment:
that's restrictive, I remember I had to remove it

Reply:
Ok let me play a little bit with `Chunk` and `qkv`, cuz the input shape of tensors for saving and loading should indeed be the same